parsers/UA.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import arrow
+import dateutil
+import requests
+
+MAP_GENERATION = {
+  'aes': 'nuclear',
+  'tec': 'other', # fossil 
+  'tes': 'coal',
+  'chpp': 'gas',
+  'hpp': 'hydro',
+  'vde': 'wind',
+  'gesgaes' : 'gesgaes', # hydro pumped storage
+  'consumptiongaespump' :'consumptiongaespump' #hydro run of river and poundage
+}
+MAP_OTHER   = {
+  'consumption' : 'consumption',
+  'hour' : 'hour'
+}
+
+tz = 'Europe/Kiev'
+
+def fetch_production(country_code='UA', session=None):
+    r = session or requests.session()
+    
+    today =  arrow.now(tz=tz).format('DD.MM.YYYY')
+    url = 'https://ua.energy/wp-admin/admin-ajax.php'
+    postdata = {
+      'action': 'get_data_oes',
+      'report_date': arrow.now(tz=tz).format('DD.MM.YYYY'),
+      'type': 'day',
+      'rnd': 0.2046847583117225
+    }
+    response = r.post(url, postdata)
+    obj = response.json()
+    allofit = []
+    for serie in obj:
+        data = []
+        data.append({
+            'countryCode': country_code,
+            'production': {},
+            'source': 'ua.energy'            
+        })
+        for key in serie:
+            key = key.encode('utf-8')
+            if key in MAP_GENERATION:
+                if key in ('gesgaes', 'consumptiongaespump'):
+                   data[0]['production']['hydro'] = serie['gesgaes'] + serie['consumptiongaespump']

```

Comment:
can the hydro pumped value be negative? You should put this value under the storage dictionary, and treat stored electricity as a positive value.